pkg/executor/wasm/handler.go

```diff
@@ -77,7 +82,37 @@ func (h *executionHandler) run(ctx context.Context) {
 		h.cancel()
 	}()
 
-	tracingEngine := tracedRuntime{h.runtime}
+	configuredProtocol := strings.ToLower(os.Getenv("OTEL_EXPORTER_OTLP_PROTOCOL"))
+	endpoint := os.Getenv("OTEL_EXPORTER_OTLP_ENDPOINT")
+
+	protocol := opentelemetry.GRPC
+	if configuredProtocol != "grpc" {
+		protocol = opentelemetry.HTTP
+	}
+
+	allowInsecure := false
+	if u, err := url.Parse(endpoint); err == nil {
+		host := u.Hostname()
+		endpoint = host + ":" + u.Port()
+		allowInsecure = host == "localhost" || host == "127.0.0.1"
+	}
+
+	conf := &opentelemetry.OTelConfig{
+		ServiceName:        "bacalhau",
+		EmitTracesInterval: time.Second * 1,
+		TraceBatchMax:      100,
+		Endpoint:           endpoint,
+		Protocol:           protocol,
+		AllowInsecure:      allowInsecure, // for localhost in dev via http
+	}
+	var adapter *opentelemetry.OTelAdapter
+	if endpoint != "" {
```

Comment:
Here, if `endpoint` is non-empty but failed to parse as a valid URL, we will still attempt to pass it to the adapter which will presumably have a bad time. We should only instantiate the adapter if the URL parse succeeded?

Reply:
Good point, it seems like re-using the client would avoid these types of errors